lib/amdResolve.js

```diff
@@ -90,10 +91,20 @@ module.exports = function resolve(pattern, {
     const fileTypePattern  = new RegExp(`\\.${fileExtension}$`);
     const extensionPattern = new RegExp(`^${targetExtension}/`);
 
+    //check if the pattern has an alias
+    const matchinAliases =  Object.entries(aliases)
+        .filter( ([alias]) => new RegExp(`^${alias}/`).test(pattern) );
```

Comment:
indexOf is useable for this simple case and it is faster.

Reply:
ok, I'll replace them I though regex where optimized by turbofan when didn't had match groups nor property change.

Reply:
https://jsperf.com/match-from-beginning-regexp-vs-indexof